code_search/code-search.ipynb

```diff
@@ -0,0 +1,451 @@
+    "  For this demo, we will use Python 2.7. This restriction is due to [Apache Beam](https://beam.apache.org/), which\n",
+    "  does not support Python 3 yet (See [BEAM-1251](https://issues.apache.org/jira/browse/BEAM-1251)).\n",
+    "\n",
+    "* `Google Cloud SDK`\n",
+    "  This example will use tools from the [Google Cloud SDK](https://cloud.google.com/sdk/). The SDK must be\n",
+    "  authenticated and authorized. See [Authentication Overview](https://cloud.google.com/docs/authentication/).\n",
+    "  \n",
+    "* `Ksonnet 0.12`\n",
+    "  We use [Ksonnet](https://ksonnet.io/) to write Kubernetes jobs in a declarative manner to be run on top of Kubeflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "echo \"Python Version Info: \" && python2 --version && echo\n",
+    "echo \"Google Cloud SDK Info: \" && gcloud --version && echo\n",
+    "echo \"Kubectl Version Info: \" && kubectl version && echo\n",
+    "echo \"Ksonnet Version Info: \" && ks version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the Source Code\n",
+    "\n",
+    "Let us clone the source code for Code Search from [kubeflow/examples](https://github.com/kubeflow/examples)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# NOTE: This cell must only be run once or the clone will complain.\n",
+    "git clone --depth=1 https://github.com/kubeflow/examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install dependencies\n",
+    "\n",
+    "Let us install all the Python dependencies. Note that everything must be done with `Python 2`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "pip2 install -r examples/code_search/src/requirements.txt\n",
+    "pip2 install https://github.com/activatedgeek/batch-predict/tarball/fix-value-provider"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Configure Variables\n",
+    "\n",
+    "This involves setting up the Ksonnet application as well as utility environment variables for various CLI steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Configuration Variables. Modify as desired.\n",
+    "\n",
+    "PROJECT = 'kubeflow-dev'\n",
+    "TARGET_DATASET = 'code_search'\n",
+    "WORKING_DIR = 'gs://kubeflow-examples/t2t-code-search/20180807'\n",
+    "\n",
+    "WORKER_MACHINE_TYPE = 'n1-highcpu-32'\n",
+    "NUM_WORKERS = 16\n",
+    "\n",
+    "# DO NOT MODIFY. The following statements are needed to support\n",
+    "# environment variables in a bash cell.\n",
```

Comment:
If you use ! notation I think can just subtitute in python variables

e.g.


!gcloud --project=${PROJECT} ....

I think that might be preferable